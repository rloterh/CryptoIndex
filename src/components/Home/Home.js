import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Container, ListGroup, Form } from 'react-bootstrap';
import { fetchCurrencies } from '../../features/currency';
import { fetchGenData } from '../../features/global';
import CurrencyCard from '../CurrencyCard/CurrencyCard';
import Hero from '../Hero/Hero';
import './Home.scss';

const Home = () => {
  const backArrow = document.querySelector('#back-arrow');
  if (backArrow !== null) {
    backArrow.classList.add('d-none');
  }
  const storeData = useSelector((state) => state);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchCurrencies());
    dispatch(fetchGenData());
  }, []);
  const currencies = storeData.currencyReducer;
  const globalData = storeData.globalReducer;
  const [currencyInput, setInput] = useState('');
  const listcurrencies = currencies.filter(
    (currency) => currencyInput === '' || currency.name.toLowerCase().includes(currencyInput) || currency.name.includes(currencyInput),
  );
  return (
    <Container>
      {globalData.map((data) => (
        <Hero
          key={data.btcDominance}
          btcDominance={data.btcDominance}
          ethDominance={data.ethDominance}
          volumeChange={data.volumeChange}
        />
      ))}
      <Form.Control
        className="search-form"
        value={currencyInput}
        onInput={(e) => setInput(e.target.value)}
        type="text"
        required
        placeholder="Search"
        id="name-input"
      />
      <div id="home-tag">
        <p>Top Cryptocurrencies</p>
      </div>
      <ListGroup data-testid="list-container">
        {listcurrencies.map((currency) => (
          <CurrencyCard
            key={currency.id}
            id={currency.id}
            name={currency.name}
            symbol={currency.symbol}
            priceUsd={currency.priceUsd}
            priceBtc={currency.priceBtc}
            mC={currency.marketCap}
            tSp={currency.totalSupply}
            rank={currency.rank}
          />
        ))}
      </ListGroup>
    </Container>
  );
};

export default Home;
