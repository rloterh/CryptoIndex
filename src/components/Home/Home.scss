.search-form {
  background-color: rgba($color: #000000, $alpha: 0.1);
  border: none;
}