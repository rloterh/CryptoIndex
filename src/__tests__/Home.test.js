import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import * as reactRedux from 'react-redux';
import renderer from 'react-test-renderer';
import Home from '../components/Home/Home';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

describe('Home', () => {
  const useSelectorMock = reactRedux.useSelector;
  const useDispatchMock = reactRedux.useDispatch;
  const mockStore = {
    currencyReducer: [
      {
        id: '1',
        name: 'BitCoin-Test',
        symbol: 'BTC-TEST',
        priceUsd: '100',
        priceBtc: '1',
        marketCap: '32000',
        totalSupply: '333',
        rank: 1,
      },
    ],
    globalReducer: [
      {
        btcDominance: '43',
        ethDominance: '19',
      },
    ],
  };
  beforeEach(() => {
    useDispatchMock.mockImplementation(() => () => {});
    useSelectorMock.mockImplementation((selector) => selector(mockStore));
  });
  afterEach(() => {
    useDispatchMock.mockClear();
    useSelectorMock.mockClear();
  });

  test('Create and show list of Currencys', () => {
    render(<Home />);
    expect(screen.getByTestId('list-container')).toBeInTheDocument();
  });

  test('Hero should render correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Home />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
