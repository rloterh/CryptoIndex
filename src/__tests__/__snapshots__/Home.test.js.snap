// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Hero should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="hero-container d-flex"
    data-testid="global-container"
  >
    <div
      className="d-data-1"
    >
      <img
        alt="logo"
        className="logo"
        src="logo.png"
      />
    </div>
    <div
      className="d-data-2"
    >
      <h2>
        ::Currency Dominance::
      </h2>
      <p>
        BTC: 
        43
        %
      </p>
      <p>
        ETH: 
        19
        %
      </p>
    </div>
  </div>
  <input
    className="search-form form-control"
    id="name-input"
    onInput={[Function]}
    placeholder="Search"
    required={true}
    type="text"
    value=""
  />
  <div
    id="home-tag"
  >
    <p>
      Top Cryptocurrencies
    </p>
  </div>
  <div
    className="list-group"
    data-testid="list-container"
    onKeyDown={[Function]}
  >
    <div
      className="list-group-item"
      data-rr-ui-event-key={null}
      data-testid="list-group-item"
      onClick={[Function]}
    >
      <div
        className="d-flex"
        id="data-container"
      >
        <div
          className="bottom-data d-flex"
        >
          <img
            alt="currency-icon"
            className="currency-image"
            src="currency.png"
          />
          <i
            className="far fa-arrow-alt-circle-right"
          />
        </div>
        <h3>
          BitCoin-Test
        </h3>
        <p>
           ( 
          BTC-TEST
           ) 
        </p>
        <p>
          $100.00
        </p>
      </div>
    </div>
  </div>
</div>
`;
